tests/unit/readers/test_skoscodelistreader.py

```diff
@@ -13,70 +13,67 @@
+    )
     assert (
         concept_uri_template
-        == "http://gss-data.org.uk/def/trade/concept/age-of-business/{+notation}"
+        == "http://gss-data.org.uk/data/gss_data/trade/ons-international-trade-in-services-by-subnational-areas-of-the-uk#concept/industry-grouping/{+notation}"
     )
 
 
-def test_skos_in_scheme_missing():
+def test_skos_concept_scheme_csvw():
     """
-    Ensure that we get an appropriate exception when the `skos:inScheme` triple column is missing.
+    Ensure that we can successfully extract the concept scheme URI and concept URI template from a roughly standard codelist CSV-W.
     """
     code_list_csvw = (
-        _skos_codelist_reader_test_cases / "skos-in-scheme-missing.csv-metadata.json"
+        _test_case_base_dir / "utils" / "csvw" / "code-list.csv-metadata.json"
     )
-
-    with pytest.raises(ValueError) as ex:
-        extract_code_list_concept_scheme_info(code_list_csvw)
-
-    assert "is missing `skos:inScheme` column" in str(ex)
-
-
-def test_about_url_missing():
-    """
-    Ensure that we get an appropriate exception when the concept aboutUrl is missing.
-    """
-    code_list_csvw = (
-        _skos_codelist_reader_test_cases / "about-url-missing.csv-metadata.json"
+    csv_path, cs_uri, concept_uri_template = extract_code_list_concept_scheme_info(
+        code_list_csvw
     )
 
-    with pytest.raises(ValueError) as ex:
-        extract_code_list_concept_scheme_info(code_list_csvw)
-
-    assert "is missing `aboutUrl` property" in str(ex)
+    assert csv_path == "code-list.csv"
+    assert cs_uri == "http://gss-data.org.uk/def/trade/concept-scheme/age-of-business"
+    assert (
+        concept_uri_template
+        == "http://gss-data.org.uk/def/trade/concept/age-of-business/{+notation}"
+    )
 
 
-def test_csv_url_missing():
+def test_num_of_variables_in_about_url_invalid_exception():
     """
-    Ensure that we get an appropriate exception when the table's URL is missing.
+    Ensures that an exception is thrown when the number of variables in about url are invalid.
     """
     code_list_csvw = (
-        _skos_codelist_reader_test_cases / "csv-url-missing.csv-metadata.json"
+        _skos_codelist_reader_test_cases
+        / "about-url-invalid-variable.csv-metadata.json"
     )
 
     with pytest.raises(ValueError) as ex:
         extract_code_list_concept_scheme_info(code_list_csvw)
 
-    assert "is missing `url` property for code list table" in str(ex)
+    assert (
+        "Unexpected number of variables in aboutUrl Template. Expected 1, found 2"
+        in str(ex)
+    )
 
 
 def test_legacy_composite_code_list():
-    """Test that a legacy composite code list returns a sensible `aboutUrl`. Addresses bug in issue #389."""
+    """Test that a legacy composite code list returns a sensible `about-url-wriUrl`. Addresses bug in issue #389."""
```

Comment:
Can we also test what happens when we fail to extract the table_url/scheme url/concept url from a CSV-W? i.e. that we get an appropriate error that would make sense to a user.

Reply:
@robons Actually, I tried to do this. But turned out that if either of these variables cannot be extracted, the sparql return 0 results which will trigger the `InvalidNumberOfRecordsException` exception from the python end.

Do you know whether sparql can do a validation and trigger a specific error (e.g. `aboutUrl` is missing) instead of returning 0 results?